### ISSUE TITLE
Inverted black and white from optimization

### ISSUE THREAD

**Issue by albertinator** [jbarlow83/OCRmyPDF#419]
**Describe the issue**
The OCR'ed output ends up with black and white colors inverted.

**To Reproduce**
I ran:
```bash
ocrmypdf -j1 --rotate-pages --deskew --clean
```
I end up getting colors inverted on the attached document.

I know this is due to optimization because if I run:
```bash
ocrmypdf  -j1 --rotate-pages --deskew --clean --optimize 0
```
then the problem doesn't come up anymore.

I do need to optimize to keep the file sizes small, and can't tell ahead of time whether or not this effect will happen at optimization.

**Example file**
An example `input.pdf` is attached for original input:
[input.pdf](https://github.com/jbarlow83/OCRmyPDF/files/3513779/input.pdf)

An example `output.pdf` is attached that is the output I get. You can clearly see the black inverted with the white when compared with `input.pdf`:
[output.pdf](https://github.com/jbarlow83/OCRmyPDF/files/3513780/output.pdf)

Please check any or all that apply about the test file:

- [ X ] This is the input file
- [ X ] The file contains no personal or confidential information
- [ ] I am the copyright holder for this file
- [ ] I permit this file to be included in the OCRmyPDF test suite under the CC-BY-SA 4.0 license
- [ ] I am not the copyright holder, but this file is available under a free software license

Files that are not free for inclusion in this project are quite welcome, but we like to collect free files for our test suite when possible. Please do *not* submit files with confidential information. At your option you may encrypt files for OCRmyPDF's author only.

**Expected behavior**
I expect no color inversion to take place

**System:**

`jbarlow83/ocrmypdf` Docker image (image ID: 1df022776970)

**Additional context**

**Comment by AEgit** [comment]
Ran into this issue as well a couple of times, but have been too lazy so far to write a bug report. So yes, I can confirm this problem.

**Comment by jbarlow83** [comment]
@AEgit @albertinator @knobix

I still haven't been able to reproduce this. All of Ghostscript, pngquant, and jbig2enc can be involved. I've attempted a fix that I think should do it. If you can pull master and give it a try, please try and see if it resolves the issue. Thanks.

**Comment by AEgit** [comment]
OS: Ubuntu 18.04 (in Windows 10 WSL)
ocrmypdf 9.0.1
tesseract 4.0.0-beta.1
gs 9.26
qpdf 8.0.2
pngquant 2.5.0

I cannot confirm that his has been fixed (assuming, that the available pip3 version is the current one). I provide another input file as example:
https://app.box.com/s/0k9foex6vy0al9kdhbeg1fj3ec1wg6bl

After OCR'ing pages 3, 5, 7, 9, and 10 have inverted colours. It is not a major problem, since the OCR'ing still seems to work - but I wanted to let you know about it.

**Comment by jbarlow83** [comment]
@AEgit You'll need to install the HEAD revision to test change. The attempted fix is not in 9.0.1

**Comment by AEgit** [comment]
@jbarlow83 Thanks for the heads-up - in that case I will wait until the fix is part of the release version, since I do not know how well the Windows 10 Windows Subsystem for Linux handles multiple installed versions of ocrmypdf (or rather: I don't know, whether it gets difficult, should there be any issues). Did you try the file I uploaded?

**Comment by knobix** [comment]
> I still haven't been able to reproduce this. All of Ghostscript, pngquant, and jbig2enc can be involved. I've attempted a fix that I think should do it. If you can pull master and give it a try, please try and see if it resolves the issue. Thanks.

OS: FreeBSD 11.2-RELEASE amd64
ocrmypdf 9.0.1@b0d9775
tesseract 4.1.0
gs 9.27
qpdf 8.4.2
pngquant 2.12.3 (with/without LCMS2)

I can confirm, that commit [b0d97753](https://github.com/jbarlow83/OCRmyPDF/commit/b0d9775343a7c04a9271fc1470d415bbb14642f3) fixes the issue. The test `test_mono_notinverted` passed and both `im.png` and `out.pdf` have black fonts on white background.

However, commit [11ef78a8](https://github.com/jbarlow83/OCRmyPDF/commit/11ef78a8912cfacb72063bccbde69826d4859d85) leads to new failures of around 52 tests:
```
E       ocrmypdf.exceptions.MissingDependencyError: Could not --print-parameters from tesseract
```

If revert that commit `11ef78a8` then all tests runs fine so far and still give the expected results.

**Comment by albertinator** [comment]
@jbarlow83 my apologies, I'm not sure how I generated that input.pdf originally, but here is one I just tested that exhibits the issue. Both in the Docker image I was using before, and in a new build (using the latest `master` pull), you should be able to reproduce the issue now:

[CCI15082019_3.pdf](https://github.com/jbarlow83/OCRmyPDF/files/3571552/CCI15082019_3.pdf)

**Comment by jbarlow83** [comment]
@albertinator This was... bewildering and surprisingly difficult for a problem that amounts to a bit flip.

I think the difference problem in platform behavior appears to be due to Leptonica commit a069230768f439299c5c9834a80f8c1e34ae005e which is in Leptonica 1.77.0 and newer (including the Homebrew macOS version I use), but not in Ubuntu 19.04 which has 1.76.0. I think that commit actually broke what it purports to fix, but so be it. Relatedly it also seems that the TIFF specification is wrong on its definition photometric interpretation for monochrome images, that or I'm losing my sanity. 

I'm now taking a difference approach to optimizing 1bpp files that seems to be resilient across Leptonica versions and does not rely on a ugly PDF hack, and it seems to work where I've checked it.

**Comment by jbarlow83** [comment]
Fixed in v9.0.2

**Comment by AEgit** [comment]
Can confirm: This does, indeed, appear to be fixed in ocrmypdf 9.0.2. Thank you very much for your work!

**Comment by albertinator** [comment]
Confirmed that this is fixed now in 9.0.2. Thank you so much for your attention to this!

**Comment by AmirthaKrishnan** [comment]
@jbarlow83 pls help me out.!"
I cant able to pull the latest version of ocrmypdf.
If i use this command - docker pull jbarlow83/ocrmypdf-alpine am in the version of **8.2.2.post10+g6e49bb3.d20190403**
How to get the latest version of ocrmypdf in docker.?
Thanks
